tiny_dnn/layers/layer.h

```diff
@@ -347,6 +353,70 @@ class layer : public node {
     return *this;
   }
 
+  /**
+   * @name Parameter Getters and Setters
+   * @{
+   */
+
+  /** @brief Add new parameter to this layer, to be called in constructor. */
+  void add_parameter(size_t width,
```

Comment:
I'd reorder parameters following the data storage model: NCHW or in this case `n_fmaps, depth, height, width`. 

Besides, I'm not really sure that this vocabulary is the best for this scope. `in_features, ?, height, width`. Take also into account that RNN might have extra dimensionality at some points, not just 4 dimensions.

@prlz77 could you confirm that?